Under some unknown and random circumstances, the minerva shell command fails with an error

This seems to happen randomly and fairly infrequently. I only mention this as I have been catching it somewhat more often. Rerunning without any changes is almost always successful.

With the command:

`./bin/minerva-cli.sh --import-owl-models -f models -j blazegraph.jnl`

After several minutes of successfully running, I occasionally get a fatal error like:

```
08:54:03  java.lang.ClassCastException
08:54:03  	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
08:54:03  	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
08:54:03  	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
08:54:03  	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
08:54:03  	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:603)
08:54:03  	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
08:54:03  	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)
08:54:03  	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
08:54:03  	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
08:54:03  	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
08:54:03  	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
08:54:03  	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:661)
08:54:03  	at org.geneontology.minerva.cli.CommandLineInterface.importOWLModels(CommandLineInterface.java:578)
08:54:03  	at org.geneontology.minerva.cli.CommandLineInterface.main(CommandLineInterface.java:251)
08:54:03  Caused by: java.lang.ClassCastException: class java.util.HashMap$Node cannot be cast to class java.util.HashMap$TreeNode (java.util.HashMap$Node and java.util.HashMap$TreeNode are in module java.base of loader 'bootstrap')
08:54:03  	at java.base/java.util.HashMap$TreeNode.moveRootToFront(HashMap.java:1900)
08:54:03  	at java.base/java.util.HashMap$TreeNode.putTreeVal(HashMap.java:2079)
08:54:03  	at java.base/java.util.HashMap.putVal(HashMap.java:634)
08:54:03  	at java.base/java.util.HashMap.put(HashMap.java:608)
08:54:03  	at java.base/java.util.HashSet.add(HashSet.java:220)
08:54:03  	at org.geneontology.minerva.cli.CommandLineInterface.lambda$importOWLModels$1(CommandLineInterface.java:591)
08:54:03  	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
08:54:03  	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
08:54:03  	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
08:54:03  	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
08:54:03  	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
08:54:03  	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
08:54:03  	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
08:54:03  	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
08:54:03  	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
08:54:03  	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

I don't think this is a particularly bad problem at this point, but I wanted to create a record for it.

--- a/minerva-cli/src/main/java/org/geneontology/minerva/cli/CommandLineInterface.java
+++ b/minerva-cli/src/main/java/org/geneontology/minerva/cli/CommandLineInterface.java
@@ -66,6 +66,8 @@ import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Paths;
 import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 import static org.geneontology.minerva.server.handler.OperationsTools.createModelRenderer;
 
@@ -568,7 +570,8 @@ public class CommandLineInterface {
         CurieHandler curieHandler = new MappedCurieHandler();
         BlazegraphMolecularModelManager<Void> m3 = new BlazegraphMolecularModelManager<>(dummy, curieHandler, modelIdPrefix, journalFilePath, null, null, false);
         //in case of update rather than whole new journal
-        Set<IRI> stored = new HashSet<IRI>(m3.getStoredModelIds());
+        Set<IRI> stored = ConcurrentHashMap.newKeySet();
+        stored.addAll(m3.getStoredModelIds());
         LOGGER.info("loading gocams from " + inputFolder);
         //for (File file : FileUtils.listFiles(new File(inputFolder), null, true)) {
         File i = new File(inputFolder);
@@ -580,15 +583,14 @@ public class CommandLineInterface {
                         java.util.Optional<String> irio;
                         try {
                             irio = m3.scanForOntologyIRI(file);
-                            IRI iri = null;
-                            if (irio.isPresent()) {
-                                iri = IRI.create(irio.get());
+                            if (!irio.isPresent()) {
+                                throw new OWLOntologyCreationException("Detected anonymous ontology; must have IRI: " + file);
                             }
+                            IRI iri = IRI.create(irio.get());
                             //is it in there already?
-                            if (stored.contains(iri)) {
+                            if (!stored.add(iri)) {
                                 LOGGER.error("Attempted to load gocam ttl file into database but gocam with that iri already exists, skipping " + file + " " + iri);
                             } else {
-                                stored.add(iri);
                                 m3.importModelToDatabase(file, true);
                             }
                         } catch (RDFParseException | RDFHandlerException | IOException e1) {
@@ -896,7 +898,7 @@ public class CommandLineInterface {
         String goshapemapFileUrl = "https://raw.githubusercontent.com/geneontology/go-shapes/master/shapes/go-cam-shapes.shapeMap";
         CurieMappings localMappings = new CurieMappings.SimpleCurieMappings(Collections.singletonMap(modelIdcurie, modelIdPrefix));
         CurieHandler curieHandler = new MappedCurieHandler(DefaultCurieHandler.loadDefaultMappings(), localMappings);
-        Map<String, String> modelid_filename = new HashMap<String, String>();
+        ConcurrentMap<String, String> modelid_filename = new ConcurrentHashMap<String, String>();
 
         if (outputFolder == null) {
             LOGGER.error("please specify an output folder with -r ");
@@ -954,17 +956,17 @@ public class CommandLineInterface {
         if (i.exists() && !input.endsWith(".jnl")) {
             if (i.isDirectory()) {
                 LOGGER.info("Loading models from " + i.getAbsolutePath());
-                Set<String> model_iris = new HashSet<String>();
                 FileUtils.listFiles(i, null, true).parallelStream().parallel().forEach(file -> {
                     if (file.getName().endsWith(".ttl") || file.getName().endsWith("owl")) {
                         try {
                             String modeluri = m3.importModelToDatabase(file, true);
                             if (modeluri == null) {
                                 LOGGER.error("Null model IRI: " + modeluri + " file: " + file);
-                            } else if (!model_iris.add(modeluri)) {
-                                LOGGER.error("Multiple models with same IRI: " + modeluri + " file: " + file + " file: " + modelid_filename.get(modeluri));
                             } else {
-                                modelid_filename.put(modeluri, file.getName());
+                                String existingFile = modelid_filename.putIfAbsent(modeluri, file.getName());
+                                if (existingFile != null) {
+                                    LOGGER.error("Multiple models with same IRI: " + modeluri + " file: " + file + " file: " + existingFile);
+                                }
                             }
                         } catch (OWLOntologyCreationException | RepositoryException | RDFParseException
                                  | RDFHandlerException | IOException e) {

